tests/test_interface.rb

```diff
@@ -1485,4 +1485,237 @@ def test_ipv4_pim_sparse_mode
+      assert_nil(interface.ipv4_dhcp_relay_subnet_broadcast)
+      assert_nil(interface.default_ipv4_dhcp_relay_subnet_broadcast)
+      assert_raises(Cisco::UnsupportedError) do
+        interface.ipv4_dhcp_relay_subnet_broadcast = true
+      end
+      return
+    end
+    assert_equal(interface.default_ipv4_dhcp_relay_subnet_broadcast,
+                 interface.ipv4_dhcp_relay_subnet_broadcast)
+    interface.ipv4_dhcp_relay_subnet_broadcast = true
+    assert_equal(true, interface.ipv4_dhcp_relay_subnet_broadcast)
+    interface.ipv4_dhcp_relay_subnet_broadcast =
+      interface.default_ipv4_dhcp_relay_subnet_broadcast
+    assert_equal(interface.default_ipv4_dhcp_relay_subnet_broadcast,
+                 interface.ipv4_dhcp_relay_subnet_broadcast)
+  end
+
+  def test_ipv4_dhcp_smart_relay
+    interface = Interface.new(interfaces[0])
+    interface.switchport_enable(false)
+    if validate_property_excluded?('interface', 'ipv4_dhcp_smart_relay')
+      assert_nil(interface.ipv4_dhcp_smart_relay)
+      assert_nil(interface.default_ipv4_dhcp_smart_relay)
+      assert_raises(Cisco::UnsupportedError) do
+        interface.ipv4_dhcp_smart_relay = true
+      end
+      return
+    end
+    assert_equal(interface.default_ipv4_dhcp_smart_relay,
+                 interface.ipv4_dhcp_smart_relay)
+    interface.ipv4_dhcp_smart_relay = true
+    assert_equal(true, interface.ipv4_dhcp_smart_relay)
+    interface.ipv4_dhcp_smart_relay =
+      interface.default_ipv4_dhcp_smart_relay
+    assert_equal(interface.default_ipv4_dhcp_smart_relay,
+                 interface.ipv4_dhcp_smart_relay)
+  end
+
+  def test_ipv6_dhcp_relay_addr
+    interface = Interface.new(interfaces[0])
+    interface.switchport_enable(false)
+    list = ['2000::11', '2000::12']
+    if validate_property_excluded?('interface', 'ipv6_dhcp_relay_addr')
+      assert_nil(interface.ipv6_dhcp_relay_addr)
+      assert_nil(interface.default_ipv6_dhcp_relay_addr)
+      assert_raises(Cisco::UnsupportedError) do
+        interface.ipv6_dhcp_relay_addr = list
+      end
+      return
+    end
+    assert_equal(interface.default_ipv6_dhcp_relay_addr,
+                 interface.ipv6_dhcp_relay_addr)
+    interface.ipv6_dhcp_relay_addr = list
+    assert_equal(list, interface.ipv6_dhcp_relay_addr)
+    interface.ipv6_dhcp_relay_addr =
+      interface.default_ipv6_dhcp_relay_addr
+    assert_equal(interface.default_ipv6_dhcp_relay_addr,
+                 interface.ipv6_dhcp_relay_addr)
+  end
+
+  def test_ipv6_dhcp_relay_src_intf
+    interface = Interface.new(interfaces[0])
+    interface.switchport_enable(false)
+    if validate_property_excluded?('interface', 'ipv6_dhcp_relay_src_intf')
+      assert_nil(interface.ipv6_dhcp_relay_src_intf)
+      assert_nil(interface.default_ipv6_dhcp_relay_src_intf)
+      assert_raises(Cisco::UnsupportedError) do
+        interface.ipv6_dhcp_relay_src_intf = 'ethernet 2/1'
```

Comment:
Don't hardcode the interface names.  Instead at the beginning of this method do this:
```
def test_ipv6_dhcp_relay_src_intf
    inf_name = interfaces[0]
    interface = Interface.new(inf_name)
```
Then use `intf_name`